tests/test_noninteractive_mode.py

```diff
@@ -32,6 +32,19 @@ def test_query(self, query_str, file_baseline):
         output_baseline = self.get_file_contents(file_baseline)
         assert output_query == output_baseline
 
+    @staticmethod
+    def test_noninteractive_run():
+        """ Test that calling run throws an exception only when interactive_mode is false """
+        mssqlcli = create_mssql_cli(interactive_mode=False)
+        try:
+            mssqlcli.run()
+            mssqlcli.shutdown()
```

Comment:
@pensivebrian I had some problems getting `shutdown()` to call in the event that `mssqlcli` actually spins up interactive mode. Is there a way to shutdown `mssqlcli` from the code?

Reply:
Run throws an exception which skips shutdown. Put run in a try statement block and shutdown in a finally statement block.